src/components/partnership/partnership.service.ts

```diff
@@ -188,6 +188,10 @@ export class PartnershipService {
       throw e;
     }
 
+    if (input.fundingType && !this.canAddFundingType(input.types)) {
+      throw new ServerException('cannot add funding type');
+    }
```

Comment:
This is not the server's fault, it's the users.
```suggestion
    if (input.fundingType && !this.canAddFundingType(input.types)) {
      throw new InputException('Funding type can only be applied to managing partners', 'partnership.fundingType');
    }
```

Reply:
Actually you could change the `canAddFundingType` function into a `verifyFundingType(fundingType, types)` function so you don't have to duplicate the exception message in both spots